### PR TITLE
Remove upper bound for pandas library

### LINKED ISSUE #349: Remove upper bound for pandas
### Problem Description

With the newly released [pandas 2.0](https://pandas.pydata.org/docs/dev/whatsnew/v2.0.0.html) it would nice to support it across the sdv-dev libraries.
 
### Expected behavior

The newly expected version for pandas should be only `>=` than the minimum supported without upper bound. After the changes are done, both minimum and newest version should work properly.

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #349

### CODE REVIEW

File: tests/unit/multivariate/test_gaussian.py [hunk 1]
```diff
@@ -423,7 +423,7 @@ def test_sample_random_state(self):
         result = instance.sample(5)
 
         # Check
-        pd.testing.assert_frame_equal(result, expected_result, check_less_precise=True)
```
Comment (amontanez24) [inline tests/unit/multivariate/test_gaussian.py:426]: did we need the `check_less_precise=True`?

Comment (pvk-developer) [reply tests/unit/multivariate/test_gaussian.py:426]: This is now deprecated, but I can do `check_exact=True` if we need to ensure exact numbers

**Review by fealho — APPROVED** [review]

File: tests/unit/multivariate/test_gaussian.py (hunk 1 above)
Comment (pvk-developer) [reply tests/unit/multivariate/test_gaussian.py:426]: Actually they suggest using `rtol` 🤔

Comment (amontanez24) [reply tests/unit/multivariate/test_gaussian.py:426]: I don't think we were trying to check exact before but actually the opposite. The parameter was giving some tolerance and I'm wondering if not doing it anymore will lead to a fragile test. Here is another possible replacement https://github.com/pandas-dev/pandas/pull/30562

Comment (amontanez24) [reply tests/unit/multivariate/test_gaussian.py:426]: But if the test passes without any issues we can just get rid of it

**Pull request merged**